File: frontend/src/services/api/network.ts
// Network API — services / ingresses / ingress classes / endpoints /
// endpoint slices / network policies. Gateway API objects (Gateway,
// HTTPRoute, GRPCRoute, ReferenceGrant, BackendTLS/TrafficPolicy)
// live in gateway.ts.

import { client } from './client'
import type {
  EndpointInfo,
  EndpointSliceInfo,
  IngressClassInfo,
  IngressDetail,
  IngressInfo,
  NetworkPolicyInfo,
  ServiceInfo,
} from './types'

export const networkApi = {
  // Services
  getServices: async (namespace: string, forceRefresh = false): Promise<ServiceInfo[]> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/services`, {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  getAllServices: async (forceRefresh = false): Promise<ServiceInfo[]> => {
    const { data } = await client.get('/cluster/services/all', {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  describeService: async (namespace: string, name: string): Promise<any> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/services/${name}/describe`)
    return data
  },

  deleteService: async (namespace: string, name: string): Promise<void> => {
    await client.delete(`/cluster/namespaces/${namespace}/services/${name}`)
  },

  // Ingresses
  getIngresses: async (namespace: string, forceRefresh = false): Promise<IngressInfo[]> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/ingresses`, {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  getAllIngresses: async (forceRefresh = false): Promise<IngressInfo[]> => {
    const { data } = await client.get('/cluster/ingresses/all', {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  getIngressClasses: async (forceRefresh = false): Promise<IngressClassInfo[]> => {
    const { data } = await client.get('/cluster/ingressclasses', {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  getIngressDetail: async (namespace: string, name: string): Promise<IngressDetail> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/ingresses/${name}/detail`)
    return data
  },

  describeIngressClass: async (name: string): Promise<IngressClassInfo> => {
    const { data } = await client.get(`/cluster/ingressclasses/${name}/describe`)
    return data
  },

  deleteIngress: async (namespace: string, name: string): Promise<void> => {
    await client.delete(`/cluster/namespaces/${namespace}/ingresses/${name}`)
  },

  deleteIngressClass: async (name: string): Promise<void> => {
    await client.delete(`/cluster/ingressclasses/${name}`)
  },

  // Endpoints
  getEndpoints: async (namespace: string, forceRefresh = false): Promise<EndpointInfo[]> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/endpoints`, {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  getAllEndpoints: async (forceRefresh = false): Promise<EndpointInfo[]> => {
    const { data } = await client.get('/cluster/endpoints/all', {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  describeEndpoint: async (namespace: string, name: string): Promise<any> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/endpoints/${name}/describe`)
    return data
  },

  deleteEndpoint: async (namespace: string, name: string): Promise<void> => {
    await client.delete(`/cluster/namespaces/${namespace}/endpoints/${name}`)
  },

  // EndpointSlices
  getEndpointSlices: async (namespace: string, forceRefresh = false): Promise<EndpointSliceInfo[]> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/endpointslices`, {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  getAllEndpointSlices: async (forceRefresh = false): Promise<EndpointSliceInfo[]> => {
    const { data } = await client.get('/cluster/endpointslices/all', {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  describeEndpointSlice: async (namespace: string, name: string): Promise<any> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/endpointslices/${name}/describe`)
    return data
  },

  deleteEndpointSlice: async (namespace: string, name: string): Promise<void> => {
    await client.delete(`/cluster/namespaces/${namespace}/endpointslices/${name}`)
  },

  // NetworkPolicies
  getNetworkPolicies: async (namespace: string, forceRefresh = false): Promise<NetworkPolicyInfo[]> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/networkpolicies`, {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  getAllNetworkPolicies: async (forceRefresh = false): Promise<NetworkPolicyInfo[]> => {
    const { data } = await client.get('/cluster/networkpolicies/all', {
      params: { force_refresh: forceRefresh },
    })
    return data
  },

  describeNetworkPolicy: async (namespace: string, name: string): Promise<any> => {
    const { data } = await client.get(`/cluster/namespaces/${namespace}/networkpolicies/${name}/describe`)
    return data
  },

  deleteNetworkPolicy: async (namespace: string, name: string): Promise<void> => {
    await client.delete(`/cluster/namespaces/${namespace}/networkpolicies/${name}`)
  },
}

File: services/k8s-service-go/internal/routes/network.go
package routes

import (
	"github.com/go-chi/chi/v5"

	"github.com/junginho0901/kubeast/services/k8s-service-go/internal/handler"
)

// RegisterNetwork — Service, Ingress, IngressClass, Endpoints,
// EndpointSlice, NetworkPolicy. Gateway-API resources live in
// gateway.go.
func RegisterNetwork(r chi.Router, h *handler.Handler) {
	// Services
	r.Get("/api/v1/services/all", h.GetAllServices)
	r.Get("/api/v1/namespaces/{namespace}/services", h.GetServices)
	r.Get("/api/v1/namespaces/{namespace}/services/{name}/describe", h.DescribeService)
	r.Get("/api/v1/namespaces/{namespace}/services/{name}/yaml", h.GetServiceYAML)
	r.Delete("/api/v1/namespaces/{namespace}/services/{name}", h.DeleteService)
	r.Get("/api/v1/namespaces/{namespace}/services/{service_name}/connectivity", h.CheckServiceConnectivity)

	// Ingresses
	r.Get("/api/v1/ingresses/all", h.GetAllIngresses)
	r.Get("/api/v1/namespaces/{namespace}/ingresses", h.GetIngresses)
	r.Get("/api/v1/namespaces/{namespace}/ingresses/{name}/describe", h.DescribeIngress)
	r.Get("/api/v1/namespaces/{namespace}/ingresses/{name}/detail", h.GetIngressDetail)
	r.Get("/api/v1/namespaces/{namespace}/ingresses/{name}/yaml", h.GetIngressYAML)
	r.Delete("/api/v1/namespaces/{namespace}/ingresses/{name}", h.DeleteIngress)

	// IngressClasses
	r.Get("/api/v1/ingressclasses", h.GetIngressClasses)
	r.Get("/api/v1/ingressclasses/{name}/describe", h.DescribeIngressClass)
	r.Delete("/api/v1/ingressclasses/{name}", h.DeleteIngressClass)

	// Endpoints
	r.Get("/api/v1/endpoints/all", h.GetAllEndpoints)
	r.Get("/api/v1/namespaces/{namespace}/endpoints", h.GetEndpoints)
	r.Get("/api/v1/namespaces/{namespace}/endpoints/{name}/describe", h.DescribeEndpoints)
	r.Get("/api/v1/namespaces/{namespace}/endpoints/{name}/yaml", h.GetEndpointsYAML)
	r.Delete("/api/v1/namespaces/{namespace}/endpoints/{name}", h.DeleteEndpoints)

	// EndpointSlices
	r.Get("/api/v1/endpointslices/all", h.GetAllEndpointSlices)
	r.Get("/api/v1/namespaces/{namespace}/endpointslices", h.GetEndpointSlices)
	r.Get("/api/v1/namespaces/{namespace}/endpointslices/{name}/describe", h.DescribeEndpointSlice)
	r.Get("/api/v1/namespaces/{namespace}/endpointslices/{name}/yaml", h.GetEndpointSliceYAML)
	r.Delete("/api/v1/namespaces/{namespace}/endpointslices/{name}", h.DeleteEndpointSlice)

	// NetworkPolicies
	r.Get("/api/v1/networkpolicies/all", h.GetAllNetworkPolicies)
	r.Get("/api/v1/namespaces/{namespace}/networkpolicies", h.GetNetworkPolicies)
	r.Get("/api/v1/namespaces/{namespace}/networkpolicies/{name}/describe", h.DescribeNetworkPolicy)
	r.Get("/api/v1/namespaces/{namespace}/networkpolicies/{name}/yaml", h.GetNetworkPolicyYAML)
	r.Delete("/api/v1/namespaces/{namespace}/networkpolicies/{name}", h.DeleteNetworkPolicy)
}
